tests/BootstrapperTest.php

```diff
@@ -326,6 +327,34 @@
     expect(File::isDirectory($tenantStoragePath))->toBeFalse();
 });
 
+test('MailTenancyBootstrapper maps tenant mail credentials to config as specified in the $credentialsMap property', function() {
+    MailTenancyBootstrapper::$credentialsMap = [
+        'mail.mailers.smtp.username' => 'smtp_username',
+        'mail.mailers.smtp.password' => 'smtp_password'
+    ];
+
+    config([
+        'mail.mailers.smtp.username' => $defaultUsername = 'default username',
+        'mail.mailers.smtp.password' => 'no password'
+    ]);
+
+    Tenant::create(['smtp_password' => 'testing password'])->run(function() use ($defaultUsername) {
+        expect(array_key_exists('smtp_password', tenant()->getAttributes()))->toBeTrue();
+        expect(array_key_exists('smtp_host', tenant()->getAttributes()))->toBeFalse();
+        expect(config('mail.mailers.smtp.username'))->toBe($defaultUsername);
+        expect(config('mail.mailers.smtp.password'))->toBe(tenant()->smtp_password);
+    });
+});
+
+test('MailTenancyBootstrapper reverts the config to default when tenancy ends', function() {
+    MailTenancyBootstrapper::$credentialsMap = ['mail.mailers.smtp.password' => 'smtp_password'];
+    config(['mail.mailers.smtp.password' => $defaultPassword = 'no password']);
+
+    Tenant::create(['smtp_password' => 'testing password'])->run(fn() => '');
```

Comment:
Make the fact that Tenancy is initialized and ended more clear. The `run()` is easy to overlook.

Use `initialize()`, then assert that the tenant password is being used, and then call `end()` and assert that we're back to the central password.

Reply:
Updated the test